utils/common/src/main/kotlin/FileMatcher.kt

```diff
@@ -39,13 +40,36 @@ class FileMatcher(
 ) {
     constructor(vararg patterns: String, ignoreCase: Boolean = false) : this(patterns.asList(), ignoreCase)
 
-    private val matcher = AntPathMatcher().apply {
-        setCaseSensitive(!ignoreCase)
+    companion object {
+        private val matchCaseInsensitive = AntPathMatcher().apply { setCaseSensitive(false) }::match
+        private val matchCaseSensitive = AntPathMatcher().apply { setCaseSensitive(true) }::match
+
+        /**
+         * Return true if [path] is matched by [pattern], false otherwise. The [path] must use '/' as separators, if it
+         * contains any.
+         */
+        fun match(pattern: String, path: String, ignoreCase: Boolean) =
```

Comment:
How about defaulting `ignoreCase` to `false` like in Kotlin's `equals`? Similar below.